mb_add_remix_credit_links.user.js

```diff
@@ -143,48 +138,53 @@ function addRemixCreditClickHandler(event) {
     const remixer = this.getAttribute('data-remixer');
     const remixOf = this.getAttribute('data-remix-of');
 
-    const addRel = recording.getElementsByClassName('add-rel')[0];
+    const addRel = recording.querySelector('button.add-relationship');
     addRel.click();
 
     if (remixer) {
         // wait 250ms for the dialog to be added to the DOM
         window.setTimeout(function () {
-            const dialog = document.getElementById('dialog');
-            const linkType = dialog.getElementsByClassName('link-type')[0];
-            setElementValue(linkType, RemixerOptionValue, 'change');
+            const dialog = document.getElementById('add-relationship-dialog');
+            const entityType = dialog.querySelector('select.entity-type');
+            setElementValue(entityType, 'artist', 'change');
+
+            const linkType = dialog.querySelector('input.relationship-type');
+            setElementValue(linkType, 'remixed / remixer');
+            tabToConfirmFirstOption(linkType);
 
-            const name = dialog.getElementsByClassName('name')[0];
+            const name = dialog.querySelector('input.relationship-target');
             if (remixer) {
                 setElementValue(name, remixer);
-            } else {
-                name.focus();
             }
+            name.focus();
         }, 250);
     } else if (remixOf) {
         // wait 250ms for the dialog to be added to the DOM
         window.setTimeout(function () {
-            const dialog = document.getElementById('dialog');
-            const entityType = dialog.getElementsByClassName('entity-type')[0];
+            const dialog = document.getElementById('add-relationship-dialog');
+            const entityType = dialog.querySelector('select.entity-type');
             setElementValue(entityType, 'recording', 'change');
 
             // wait another 250ms for the link-type select options to be updated
             window.setTimeout(function () {
-                const linkType = dialog.getElementsByClassName('link-type')[0];
-                setElementValue(linkType, RemixOfOptionValue, 'change');
+                const linkType = dialog.querySelector('input.relationship-type');
+                setElementValue(linkType, 'remix of / has remixes');
+                tabToConfirmFirstOption(linkType);
 
-                const name = dialog.getElementsByClassName('name')[0];
+                const name = dialog.querySelector('input.relationship-target');
                 if (remixOf) {
                     setElementValue(name, remixOf);
-                } else {
-                    name.focus();
                 }
+                name.focus();
```

Comment:
As per my comment above, `if (remixOf)` seems redundant and the following change resolved the search issue for me:

```js
                const search = name.parentElement.querySelector('button.search');
                if (search) {
                    window.setTimeout(function() {
                        search.click();
                        name.focus();
                    }, 100);
                }
```